addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -37,21 +42,24 @@ const PopupWidget = publicWidget.Widget.extend({
      * @private
      */
     _bindPopup: function () {
-        const $main = this.$target.find('.s_popup_main');
-
-        let display = $main.data('display');
-        let delay = $main.data('showAfter');
 
         if (config.device.isMobile) {
-            if (display === 'onExit') {
-                display = 'afterDelay';
-                delay = 5000;
+            if (this.display === 'onExit') {
+                this.display = 'afterDelay';
+                this.delay = 5000;
             }
             this.$('.s_popup_main').removeClass('s_popup_center').addClass('s_popup_bottom');
         }
 
-        if (display === 'afterDelay') {
-            this.timeout = setTimeout(() => this._showPopup(), delay);
+        if (this.display === 'afterDelay') {
+            this.timeout = setTimeout(() => this._showPopup(), this.delay);
+        } else if (this.display == 'onClick') {
+            $(document).ready(function () {
+                $("a[href$='#" + this.$el.attr('id') + "']").click(function (ev) {
+                    ev.preventDefault();
+                    this._showPopup();
+                }.bind(this));
+            }.bind(this));
```

Comment:
You won't need this, Bootstrap has an automatic way to link a link to a modal. You should not have to modify this code but instead the link dialog code to add `data-toggle="modal"` when necessary.

Other reasons this is not ok:
- You'll never need document ready. Here you are in a public widget which is automatically instantiated on page load... and this instantiation occurs after DOM searches etc, which are made after the dom is ready. Also... our JS is lazy loaded and we start lazy loading once everything else is loaded ;)
- This is very rare to have to attach event handlers this way (most of the time, the widget `events` key should be used) and when you do need this, you have to unbind it somewhere (the destroy method). Otherwise, the event handler will persist when entering edit mode while it should not, or it will be duplicated as the widget are restarted when entering edit mode.